drivers/serial/uart_mcux_lpuart.c

```diff
@@ -1179,6 +1179,40 @@ static int mcux_lpuart_configure(const struct device *dev,
 }
 #endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
 
+#if CONFIG_UART_LINE_CTRL &&  \
+	defined(FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT) && FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT
+static int mcux_lpuart_line_ctrl_set(const struct device *dev,
+		uint32_t ctrl, uint32_t val)
+{
+	const struct mcux_lpuart_config *config = dev->config;
+	int ret = 0;
+
+	switch (ctrl) {
+	case UART_LINE_CTRL_RTS:
+		/* Disable Transmitter and Receiver */
+		config->base->CTRL &= ~(LPUART_CTRL_TE_MASK | LPUART_CTRL_RE_MASK);
+
+		if (val >= 1U) {
+			/* Reset TXRTS to set RXRTSE bit, this provides high-level on RTS line */
+			config->base->MODIR &= ~(LPUART_MODIR_TXRTSPOL_MASK |
+									 LPUART_MODIR_TXRTSE_MASK);
+			config->base->MODIR |= LPUART_MODIR_RXRTSE_MASK;
+		} else {
+			/* Set TXRTSE to reset RXRTSE bit,this provide low-level on RTS line*/
+			config->base->MODIR &= ~(LPUART_MODIR_RXRTSE_MASK);
+			config->base->MODIR |= (LPUART_MODIR_TXRTSPOL_MASK |
+									LPUART_MODIR_TXRTSE_MASK);
+		}
+		break;
+
+	default:
+		ret = -ENODEV;
```

Comment:
should be -ENOTSUP according to API if some control option is not enabled I think